python/cudf/cudf/tests/test_datetime.py

```diff
@@ -1551,3 +1551,34 @@ def test_error_values():
         match="DateTime Arrays is not yet implemented in cudf",
     ):
         s.values
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            [
+                "2020-05-31 08:00:00",
+                "1999-12-31 18:40:10",
+                "2000-12-31 04:00:05",
+                "1900-02-28 07:00:06",
+                "1800-03-14 07:30:20",
+                "2100-03-14 07:30:20",
+                "1970-01-01 00:00:09",
+                "1969-12-31 12:59:10",
+            ]
+        )
+    ],
+)
+def test_ceil(data):
+    resolutions = ["D", "H", "T", "S", "L", "U", "N"]
+
+    for time_type in DATETIME_TYPES:
+        ps = pd.Series(data, dtype=time_type)
+        gs = cudf.from_pandas(ps)
+
+        for resolution in resolutions:
```

Comment:
my only nit would be that I think it's a little easier to debug if we just parameterize over all these things.

Reply:
so write individual testcases for each resolution?